<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Promise</title>
</head>
<body>
  <script>
    // const promise = new Promise(function(resolve, reject) {
    //   /**
    //    * resolve函数的作用是，将Promise对象的状态从“未完成”变为“成功”（即从 pending 变为 resolved），在异步操作成功时调用，并将异步操作的结果，作为参数传递出去；
    //    * reject函数的作用是，将Promise对象的状态从“未完成”变为“失败”（即从 pending 变为 rejected），在异步操作失败时调用，并将异步操作报出的错误，作为参数传递出去
    //   */
    //   isAsync = setTimeout(() => {},1000)
    //   if (isAsync){
    //     resolve({a:1}); // 会将参数{a:1}传出去
    //     // resolve() // 结果会是undefined，传递出去的值也会是undefined
    //   } else {
    //     // reject({a:'err'}); // 会将参数{a:'err'}传出去
    //     reject(); // 结果会是undefined，传递出去的值也会是undefined
    //   }
    // });
    // const p = promise.then(res => {
    //   console.log(res, 'succ') // res则是resolve(value)中value的值，没有传value则是undefined
    // }).catch(err => {
    //   console.log(err, 'err') // err则是reject(error)中error的值，没有传error则是undefined
    // })

    function testResolve() {
      console.log(1)
      const fool = Promise.resolve('foo')
      const cc = fool.then(res => {
        console.log(res)
      })
      console.log(2)
      return fool
      // 1 2 fool
    }
    const result = testResolve()
    console.log(result, 'result')
  </script>
</body>
</html>